### PR TITLE
Add flag to ignore partition key type check

### PR DESCRIPTION
**Pull request by felix-weizman-deel**
Hello,

We found a few cases when the type checking is doesn't work for us.
The easiest solution will be to be able ignore the validation in this cases.

What do you think?

Best,
Felix

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -7,7 +7,7 @@ def rel(*xs):
     return os.path.join(os.path.abspath(os.path.dirname(__file__)), *xs)
 
 
-with open(rel("README.rst")) as f:
+with open(rel("README.rst"), encoding="utf8", errors='ignore') as f:
```
Comment (felix-weizman-deel) [inline setup.py:10]: Solve problem in windows with:
```
return codecs.charmap_decode(input,self.errors,decoding_table)[0]
    UnicodeDecodeError: 'charmap' codec can't decode byte 0x81
```

**Review by idoki-boxycharm — COMMENTED** [review]
definitely something I'd need

File: pg_partitioning/decorators.py [hunk 2]
```diff
@@ -2,14 +2,16 @@
...
 
 from django.db import models
+
 from pg_partitioning.manager import ListPartitionManager, TimeRangePartitionManager
 
 logger = logging.getLogger(__name__)
 
 
 class _PartitioningBase:
-    def __init__(self, partition_key: str, **options):
```
Comment (virusdefender) [inline pg_partitioning/decorators.py]: could you add some document to describe when we should ignore the check

**Review by virusdefender — CHANGES_REQUESTED** [review]

File: pg_partitioning/decorators.py (hunk 2 above)
Comment (felix-weizman-deel) [reply pg_partitioning/decorators.py]: I add one example, a partitioning by ForeignKey
I also changed added check_partition_key_type only to ListPartitioning

Comment (felix-weizman-deel) [reply pg_partitioning/decorators.py]: @virusdefender any updates?

**Review by virusdefender — APPROVED** [review]

**Comment by virusdefender** [comment]
it seems the ci test failed, i retried it, waiting for the result

**Comment by monouno** [comment]
There are already better Django plugins (such as django-postgres-extra) that make it easy for Django-based projects to use the advanced features of PostgreSQL databases, so this project is no longer maintained. You can still fork this project and do secondary development, have a nice life :)

**Pull request closed without merging**